plugin/core/types.py

```diff
@@ -89,39 +89,124 @@ def cancel_pending(self) -> None:
             self._current_id = -1
 
 
+def read_bool_setting(settings_obj: sublime.Settings, key: str, default: bool) -> bool:
+    val = settings_obj.get(key)
+    if isinstance(val, bool):
+        return val
+    else:
+        return default
+
+
+def read_int_setting(settings_obj: sublime.Settings, key: str, default: int) -> int:
+    val = settings_obj.get(key)
+    if isinstance(val, int):
+        return val
+    else:
+        return default
+
+
+def read_dict_setting(settings_obj: sublime.Settings, key: str, default: dict) -> dict:
+    val = settings_obj.get(key)
+    if isinstance(val, dict):
+        return val
+    else:
+        return default
+
+
+def read_list_setting(settings_obj: sublime.Settings, key: str, default: list) -> list:
+    val = settings_obj.get(key)
+    if isinstance(val, list):
+        return val
+    else:
+        return default
+
+
+def read_str_setting(settings_obj: sublime.Settings, key: str, default: str) -> str:
+    val = settings_obj.get(key)
+    if isinstance(val, str):
+        return val
+    else:
+        return default
+
+
 class Settings:
 
-    def __init__(self) -> None:
-        self.show_view_status = True
-        self.auto_show_diagnostics_panel = 'always'
-        self.auto_show_diagnostics_panel_level = 2
-        self.diagnostics_panel_include_severity_level = 4
-        self.show_diagnostics_count_in_view_status = False
-        self.show_diagnostics_in_view_status = True
-        self.show_diagnostics_severity_level = 2
-        self.only_show_lsp_completions = False
-        self.diagnostics_highlight_style = "underline"
-        self.document_highlight_style = "stippled"
-        self.document_highlight_scopes = {
+    # This is only for mypy
+    show_view_status = None  # type: bool
```

Comment:
These types seem wrong as the value is `None` initially so the matching type would be `Optional[bool]`. `pyright` seems to agree on that but I guess mypy is fine as linter looks happy...

Reply:
That syntax is for python2, but it also works for Sublime's python3.3. I'm not sure who is right: mypy or pyright.

Reply:
Mypy documentation says:

> Note that attribute definitions in the class body that use a type comment are special: a None value is valid as the initializer, even though the declared type is not optional. This should be used sparingly, as this can result in None-related runtime errors that mypy can’t detect.

https://mypy.readthedocs.io/en/stable/class_basics.html#class-attribute-annotations

It seems like a mypy limitation since it basically says itself that this can result in runtime errors.